src/VaultClient.ts

```diff
@@ -122,9 +128,11 @@ export class VaultClient extends Account {
    */
   async depositCollateral(params: { amount: number; vault: Vault }): Promise<boolean> {
     await this.initialiseReachAccount();
-    const ctc = this.reachAccount.contract(backend, params.vault.id);
+    const ctc = this.reachAccount.contract(this.backend, params.vault.id);
     const put = ctc.a.VaultOwner;
-    const res = await put.depositCollateral(convertToMicroUnits(params.amount));
+    const res = await put.depositCollateral(
+      convertToMicroUnits(params.amount, this.asaVault ? this.asaVault.decimals : undefined),
+    );
```

Comment:
```suggestion
    const res = await put.depositCollateral(
      this.asaVault ? convertToMicroUnits(params.amount, this.asaVault.decimals : 
      convertToMicroUnits(params.amount),
    );
```
I think this is a bit easier to read, and a bit more certain in the outcome. `undefined` is odd to pass in and could lead to unexpected behavior.

Reply:
hmm I am a bit torn on this actually given how often we do it. Would appreciate input from other team members on this

Reply:
@AwolDes 
I passed that in because of null, a person could pass null in place of decimals and the code will error. let me know if you want me to change this

Reply:
I think it is ok for now, maybe something we can refactor in the future